### DISCUSSION TITLE
Local time error: IndeterminateOffset

### DISCUSSION THREAD

**Discussion by incker** [time-rs/time#389] [Question]
Trying to do
`UtcOffset::current_local_offset().unwrap();`

Cause panic

`thread 'main' panicked at 'called Result::unwrap() on an Err value: IndeterminateOffset', src/main.rs:264:58`

How can I fix it?

**Comment by jhpratt** [comment]
You're calling `.unwrap()` which is causing a panic. That's 100% on you.

Without other information, all I can do is guess as to why the `Err` value is being returned. Most likely you're on a Unix-based OS, which has soundness issues. The method is fallible for a reason.

**Comment by incker** [comment]
linux mint, if this helps
hour offset +02:00

**Reply by avl** [reply]
You probably need to enable the 'unsound_local_offset' feature. The reason is that it has (relatively recently) been discovered that obtaining the local time zone offset on linux cannot be done without creating a security risk.

More reading:
[CVE-2020-26235 ](https://nvd.nist.gov/vuln/detail/CVE-2020-26235)
https://www.reddit.com/r/rust/comments/qamgyh/is_the_chrono_crate_unmaintained/

**Reply by jhpratt** [reply]
See #293. Obtaining the local offset on Linux is unsound and so the call returns an `Err`.

**Comment by emabee** [comment]
Interestingly, maybe, `chrono::Local::now()` in contrast works as expected.

**Reply by jhpratt** [reply]
Yes, but its call is unsound and could crash your program without notice.
